model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/userSession/client/entity/JpaClientSessionEntity.java

```diff
@@ -0,0 +1,319 @@
+    }
+
+    @Override
+    public String getRedirectUri() {
+        return metadata.getRedirectUri();
+    }
+
+    @Override
+    public void setRedirectUri(String redirectUri) {
+        metadata.setRedirectUri(redirectUri);
+    }
+
+    @Override
+    public Long getTimestamp() {
+        if (isMetadataInitialized()) return metadata.getTimestamp();
+        return timestamp;
+    }
+
+    @Override
+    public void setTimestamp(Long timestamp) {
+        metadata.setTimestamp(timestamp);
+    }
+
+    @Override
+    public Long getExpiration() {
+        if (isMetadataInitialized()) return metadata.getExpiration();
+        return expiration;
+    }
+
+    @Override
+    public void setExpiration(Long expiration) {
+        metadata.setExpiration(expiration);
+    }
+
+    @Override
+    public String getAction() {
+        if (isMetadataInitialized()) return metadata.getAction();
+        return action;
+    }
+
+    @Override
+    public void setAction(String action) {
+        metadata.setAction(action);
+    }
+
+    @Override
+    public String getCurrentRefreshToken() {
+        return metadata.getCurrentRefreshToken();
+    }
+
+    @Override
+    public void setCurrentRefreshToken(String currentRefreshToken) {
+        metadata.setCurrentRefreshToken(currentRefreshToken);
+    }
+
+    @Override
+    public Integer getCurrentRefreshTokenUseCount() {
+        return metadata.getCurrentRefreshTokenUseCount();
+    }
+
+    @Override
+    public void setCurrentRefreshTokenUseCount(Integer currentRefreshTokenUseCount) {
+        metadata.setCurrentRefreshTokenUseCount(currentRefreshTokenUseCount);
+    }
+
+    @Override
+    public Boolean isOffline() {
+        if (isMetadataInitialized()) return metadata.isOffline();
+        return offline;
+    }
+
+    @Override
+    public void setOffline(Boolean offline) {
+        metadata.setOffline(offline);
+    }
+
+    @Override
+    public Map<String, String> getNotes() {
+        return Collections.unmodifiableMap(notes.stream().collect(Collectors.toMap(JpaClientSessionNoteEntity::getName, JpaClientSessionNoteEntity::getValue)));
+    }
+
+    @Override
+    public void setNotes(Map<String, String> notes) {
+        this.notes.clear();
+        if (notes == null) return;
+        for (Map.Entry<String, String> entry : notes.entrySet()) {
+            setNote(entry.getKey(), entry.getValue());
+        }
+    }
+
+    @Override
+    public String getNote(String name) {
+        return notes.stream()
+                .filter(obj -> Objects.equals(obj.getName(), name))
+                .findFirst()
+                .map(JpaClientSessionNoteEntity::getValue)
+                .orElse(null);
+    }
+
+    @Override
+    public Boolean removeNote(String name) {
+        return notes.removeIf(obj -> Objects.equals(obj.getName(), name));
+    }
+
+    @Override
+    public void setNote(String name, String value) {
+        if (name == null || value == null || value.trim().isEmpty()) return;
```

Comment:
FTR: This has to be updated to reove the note.

Reply:
@hmlnarik do you mean that, if the value being passed is `null`,  this should remove the note before returning? Given there's a `removeNote` method, isn't it reasonable to expect that if anyone intends on removing a note they should use the proper method instead of expecting the `setNote` to do the removal?

Reply:
Got it, thank you!